src/modules/ekf2/EKF2.cpp

```diff
@@ -416,13 +419,20 @@ void EKF2::Run()
 
 				if ((_vehicle_status.arming_state != vehicle_status_s::ARMING_STATE_ARMED) && (_invalid_mag_id_count > 100)) {
 					// the sensor ID used for the last saved mag bias is not confirmed to be the same as the current sensor ID
-					// this means we need to reset the learned bias values to zero
+					// this means we need to reset the learned bias values to zero and
+					// the learned bias variance to a default value
 					_param_ekf2_magbias_x.set(0.f);
 					_param_ekf2_magbias_x.commit_no_notification();
 					_param_ekf2_magbias_y.set(0.f);
 					_param_ekf2_magbias_y.commit_no_notification();
 					_param_ekf2_magbias_z.set(0.f);
 					_param_ekf2_magbias_z.commit_no_notification();
+					_param_ekf2_magb_var_x.set(2.5e-7f);
```

Comment:
We could add a reset (to default) instead. https://github.com/PX4/Firmware/blob/master/platforms/common/include/px4_platform_common/param.h

Reply:
That would be better, yes. Could you help me to do that "reset to default" function please?